src/main/java/com/marklogic/spark/reader/filter/FilterFactory.java

```diff
@@ -0,0 +1,59 @@
+package com.marklogic.spark.reader.filter;
+
+import org.apache.parquet.filter2.predicate.Operators;
+import org.apache.spark.sql.sources.*;
+
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FilterFactory {
```

Comment:
This is the thing that knows how to convert a Spark Filter into an "OpticFilter", which then knows how to modify the user's serialized Optic query and bind a param value if necessary.